plugin/formatting.py

```diff
@@ -119,6 +120,7 @@ def is_enabled(self, event: Optional[dict] = None, point: Optional[int] = None)
 
     def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
         session = self.best_session(self.capability)
-        if session:
-            req = text_document_range_formatting(self.view, self.view.sel()[0])
+        selection = first_selection(self.view)
```

Comment:
Strictly speaking, the Selection is a container of regions, here we take the first region, not the selection.